docs/preview/02-Features/02-message-handling/01-service-bus.md

```diff
@@ -140,111 +131,105 @@ public class Startup
-                options.JobId = Guid.NewGuid().ToString();
+The registration of these message body handlers can be done just as easily as an `IMessageSerializer`:
 
-                // The name of the Azure Service Bus message property that has the transaction ID.
-                // (default: Transaction-Id).
-                options.Correlation.TransactionIdPropertyName = "X-Transaction-ID";
+```csharp
+using Microsoft.Extensions.DependencyInjection;
 
-                // Indicate whether or not the default built-in JSON deserialization should ignore additional members 
-                // when deserializing the incoming message (default: AdditionalMemberHandling.Error).
-                options.Deserialization.AdditionalMembers = AdditionalMemberHandling.Ignore;
+public class Startup
+{
+    public void ConfigureServices(IServiceCollection services)
+    {
+        // Register the message body serializer in the dependency container where the dependent services will be injected.
+        services.WitServiceBusMessageHandler<OrderBatchMessageHandler>(..., messageBodySerializer: new OrderBatchMessageBodySerializer());
 
-                // Indicate whether or not a new Azure Service Bus Topic subscription should be created/deleted
-                // when the message pump starts/stops (default: CreateOnStart & DeleteOnStop).
-                options.TopicSubscription = TopicSubscription.CreateOnStart | TopicSubscription.DeleteOnStop;
-            });
+        // Register the message body serializer  in the dependency container where the dependent services are manually injected.
+        services.WithServiceMessageHandler(..., messageBodySerializerImplementationFactory: serviceProvider => 
+        {
+            var logger = serviceProvider.GetService<ILogger<OrderBatchMessageHandler>>();
+            return new OrderBatchMessageHandler(logger);
+        });
+    }
+}
+```
 
-        services.AddServiceBusQueueMessagePump(
-            "ARCUS_SERVICEBUS_ORDERS_CONNECTIONSTRING",
-            options => 
-            {
-                // Indicate whether or not messages should be automatically marked as completed 
-                // if no exceptions occurred and processing has finished (default: true).
-                options.AutoComplete = true;
+> Note that the order in which the message handlers are registered is important in the message processing.
+> In the example, when a message handler above this one is registered that could also handle the message (same message type) than that handler may be chosen instead of the one with the specific filter.
 
-                // Indicate whether or not the message pump should emit security events (default: false).
-                options.EmitSecurityEvents = true;
+### Filter messages based on message body
+When registering a new message handler, one can opt-in to add a filter on the incoming message body which filters out messages that are not needed to be processed.
```

Comment:
Can those messages that are filtered out, still be processed by another message handler ?

Reply:
Yes, of course, this is just for message routing, not for the actual processing.

Reply:
Done in https://github.com/arcus-azure/arcus.messaging/pull/268/commits/8c5291057252421fb3e696bc84e314c7e9187def